File: parking-domain/src/main/java/com/chatchatabc/parking/domain/model/File.java
package com.chatchatabc.parking.domain.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "files")
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode(callSuper = true)
public class File extends FlagEntity {
    public static int DELETED = 0;

    @Id
    private String id;

    @JsonIgnore
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "uploaded_by")
    private User uploadedBy;

    @Column
    private String parentId;

    @Column
    private String mimetype;

    @Column
    private int fileOrder;

    @JsonIgnore
    public boolean isDeleted() {
        return this.getBitValue(DELETED);
    }

    public void setDeleted(boolean value) {
        this.setBitValue(DELETED, value);
    }

    @CreationTimestamp
    private LocalDateTime createdAt;
}

File: backend/parking-service/src/main/kotlin/com/chatchatabc/service/domain/model/File.kt
package com.chatchatabc.service.domain.model

import com.fasterxml.jackson.annotation.JsonIgnore
import jakarta.persistence.*
import lombok.Data
import org.hibernate.annotations.CreationTimestamp
import java.util.*

@Data
@Entity
@Table(name = "files")
open class File {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    open lateinit var id: String

    @JsonIgnore
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "uploaded_by")
    open lateinit var uploadedBy: User

    @Column
    open lateinit var parentId: String

    @Column
    open lateinit var filename: String

    @Column
    open lateinit var mimetype: String

    @Column
    open var fileOrder: Int = 0

    @CreationTimestamp
    open lateinit var createdAt: Date
}